scripts/maxmind-db-downloader.js

```diff
@@ -0,0 +1,126 @@
+var request = require('request');
+var zlib = require('zlib');
+
+// set up mozlog, default is `heka`
+mozlog.config({
+  app: 'fxa-geodb'
+});
+var log = mozlog();
+
+var MaxmindDbDownloader = function () {
+  'use strict';
+
+  this.createTargetDir = function (targetDirName) {
+    targetDirName = targetDirName || DEFAULTS.DEFAULT_TARGET_DIR_NAME;
+    var targetDirPath = path.join(__dirname, '..', targetDirName);
+    // create db folder
+    mkdirp.sync(targetDirPath);
+    // log.info only if not testing
+    // log.error always
+    if (process.env.NODE_ENV !== 'test') {
+      log.info('Download folder is ' + targetDirPath);
+    }
+    return targetDirPath;
+  };
+
+  this.setupDownloadList = function (sourceFilePath, targetDirPath) {
+    sourceFilePath = sourceFilePath || DEFAULTS.DEFAULT_SOURCE_FILE_NAME;
+    targetDirPath = targetDirPath || DEFAULTS.DEFAULT_TARGET_DIR_PATH;
+    // import the list of files to download
+    var sources = require(sourceFilePath);
+    var remainingDownloads = [];
+
+    // push each file-load-function onto the remainingDownloads queue
+    for (var source in sources) {
+      var url = sources[source];
+      if (process.env.NODE_ENV !== 'test') {
+        log.info('Adding ' + url);
+      }
+      // get the file name without the extension
+      var targetFileName = path.parse(source).name;
+      var targetFilePath = path.join(targetDirPath, targetFileName);
+      remainingDownloads.push(this.download(url, targetFilePath));
+      if (process.env.NODE_ENV !== 'test') {
+        log.info('Setting ' + targetFilePath + ' as target file');
+      }
+    }
+    return remainingDownloads;
+  };
+
+  this.download = function (url, targetFilePath) {
+    // closure to separate multiple file-download
+    return function downloadFunctor(callback) {
+      var stream = request(url);
+      stream.pipe(zlib.createGunzip()).pipe(fs.createWriteStream(targetFilePath)).on('finish', function (err) {
+        // forces overwrite, even if file exists already
+        if (err) {
+          log.error(err);
+        } else {
+          // extraction is complete
+          if (process.env.NODE_ENV !== 'test') {
+            log.info('unzip complete');
+          }
+          callback();
+        }
+      });
+    };
+  };
+
+  this.startDownload = function (remainingDownloads) {
+    async.parallel(remainingDownloads, function (err) {
+      if (err) {
+        log.error(err);
+      } else {
+        if (process.env.NODE_ENV !== 'test') {
+          log.info('Downloads complete');
+        }
+      }
+    });
+    if (process.env.NODE_ENV !== 'test') {
+      log.info('Last Update: ', new Date());
+    }
+  };
+
+  this.setupAutoUpdate = function (cronTiming, remainingDownloads, timeZone) {
+    cronTiming = cronTiming || DEFAULTS.DEFAULT_CRON_TIMING;
+    timeZone = timeZone || DEFAULTS.DEFAULT_TIMEZONE;
+    var self = this;
+    // by default run periodically on Wednesday at 01:30:30.
```

Comment:
@vladikoff We still default to it, the DEFAULT_CRON_TIMING is set to wednesdays at 1.30am